### PR TITLE
Add docker-compose setup

### PR DESCRIPTION
**Pull request by glensc**
This is my attempt on #231.

Uses docker layered builds to optimize caching, the resulting image will be smallest, but layers caching is efficinent.

besides that, uses official docker PHP alpine image: `php:7.3-fpm-alpine` and same for mongo image. I've used percona variant but can switch to MongoDB if needed.

The customizations, for now, just edit your local `docker-compose.yml` or create `docker-compose.override.yml` with overrides. But could be extended with `.env` overrides in separate PR if there's interest.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+COPY --from=source /app/composer.* ./
+COPY --from=source /app/vendor ./vendor
+
+# install in two steps to cache composer run based on composer.* files
+RUN composer require --update-no-dev --no-scripts alcaeus/mongo-php-adapter ^1.1
+RUN composer install $COMPOSER_FLAGS --no-scripts --no-autoloader
+
+# copy rest of the project. copy in order that is least to most changed
+COPY --from=source /app/external ./external
```
Comment (markstory) [inline Dockerfile]: If xhgui is running in isolation external isn't really required, as folks would be using the xhgui-collector package.

Comment (glensc) [reply Dockerfile]: indeed. removed

**Comment by lauripiisang** [comment]
Please add docker usage information to README as well.

File: README.md [hunk 2]
```diff
@@ -85,6 +85,23 @@ Installation
...
+Installation with Docker
+========================
+
+This setup uses [docker-compose] to orchestrate docker containers.
+
+1. Clone or download `xhgui` from GitHub.
+
+2. Startup the containers: `docker-compose up -d`
+
+3. Open your browser at http://xhgui.127.0.0.1.xip.io:27017 or just http://localhost:27017
```
Comment (lauripiisang) [inline README.md]: doesn't this point the browser to the MongoDB instance? What woudl be the point of that?

Comment (glensc) [reply README.md]: too soon! pushed update!

**Comment by glensc** [comment]
added readme section.

I omitted the configuration readme block, as I intended to make it automatic:
- https://github.com/glensc/xhgui/commits/docker-mongo-init

But this needs to be tested first, I will not push to this branch until tested, so this could be still merged.

**Comment by glensc** [comment]
ok, tested the mongodb indexes init. so adding commits to this branch.

```
➔ docker-compose exec mongo mongo xhprof --eval 'db.results.getIndexes()'
Percona Server for MongoDB shell version v3.6.11-3.1
connecting to: mongodb://127.0.0.1:27017/xhprof?gssapiServiceName=mongodb
Implicit session: session { "id" : UUID("c8da7aa9-35ce-4141-a5da-2aa6c3c55c62") }
Percona Server for MongoDB server version: v3.6.11-3.1
[
	{
		"v" : 2,
		"key" : {
			"_id" : 1
		},
		"name" : "_id_",
		"ns" : "xhprof.results"
	},
	{
		"v" : 2,
		"key" : {
			"meta.SERVER.REQUEST_TIME" : -1
		},
		"name" : "meta.SERVER.REQUEST_TIME_-1",
		"ns" : "xhprof.results"
	},
	{
		"v" : 2,
		"key" : {
			"profile.main().wt" : -1
		},
		"name" : "profile.main().wt_-1",
		"ns" : "xhprof.results"
	},
	{
		"v" : 2,
		"key" : {
			"profile.main().mu" : -1
		},
		"name" : "profile.main().mu_-1",
		"ns" : "xhprof.results"
	},
	{
		"v" : 2,
		"key" : {
			"profile.main().cpu" : -1
		},
		"name" : "profile.main().cpu_-1",
		"ns" : "xhprof.results"
	},
	{
		"v" : 2,
		"key" : {
			"meta.url" : 1
		},
		"name" : "meta.url_1",
		"ns" : "xhprof.results"
	}
]
```

File: README.md [hunk 3]
```diff
@@ -85,6 +85,23 @@ Installation
...
+Installation with Docker
+========================
+
+This setup uses [docker-compose] to orchestrate docker containers.
+
+1. Clone or download `xhgui` from GitHub.
+
+2. Startup the containers: `docker-compose up -d`
+
+3. Open your browser at http://xhgui.127.0.0.1.xip.io:8142 or just http://localhost:8142
```
Comment (markstory) [inline README.md]: What is `xip.io`?

Comment (glensc) [reply README.md]: http://xip.io/

```
wildcard DNS for everyone 
    
What is xip.io?

xip.io is a magic domain name that provides wildcard DNS
for any IP address. Say your LAN IP address is 10.0.0.1.
Using xip.io,

          10.0.0.1.xip.io   resolves to   10.0.0.1
      www.10.0.0.1.xip.io   resolves to   10.0.0.1
   mysite.10.0.0.1.xip.io   resolves to   10.0.0.1
  foo.bar.10.0.0.1.xip.io   resolves to   10.0.0.1
```

Comment (markstory) [reply README.md]: Neat! thanks for sharing.

**Comment by glensc** [comment]
thanks! :)

**Pull request merged**